src/condor_tools/htcondor_cli/convert_ad.py

```diff
@@ -0,0 +1,127 @@
+from htcondor._utils.ansi import Color, colorize
+from time import time
+
+DAEMON_HP_DEFAULT = -1
+DAEMON_MP_MAX = 100
+
+
+def health_str(hp: int, num_daemons: int = 1) -> str:
+    ceil = DAEMON_MP_MAX * num_daemons
+    if hp / ceil < 0.25:
+        health = colorize("Poor", Color.BRIGHT_RED)
+    elif hp / ceil < 0.75:
+        health = colorize("Decent", Color.BRIGHT_YELLOW)
+    else:
+        health = colorize("Good", Color.BRIGHT_GREEN)
+
+    return health
+
+
+def duty_cycle_to_health(worth: int, duty_cycle: float) -> int:
+    """Convert Recent DaemonCore Duty Cycle (rdcdc) into health"""
+
+    # This function returns the HP worth * health percentage.
+    # Health percentage is calculated by 2 linear functions
+    # with an intersect @ (0.95, 0.95) since 0.95 is still
+    # healthy for the rdcdc, but 0.98+ is not healthly
+    # rdcdc <= 0.95: f(x) = -(0.05/0.95)x + 1
+    # rdcdc > 0.95:  f(x) = -(0.95/0.03)x + 31.033
+    # - Cole Bollig 2024-08-16
+
+    duty_cycle = sorted([0.0, duty_cycle, 1.0])[1]
+    return int(
+        worth
+        * max(
+            -(0.05 / 0.95 * duty_cycle) + 1
+            if duty_cycle <= 0.95
+            else -(0.95 / 0.03 * duty_cycle) + 31.033,
+            0,
+        )
```

Comment:
This whole block makes the HP thing look overly-complicated.

If you're going for a mix of numeric weights, consider replacing the explicit scaling factor here (the HP parameter) with an explicit scaling factor and a table:

```
DAEMON_HEALTH_TABLE = {
    "MASTER": [
        (100, getRecentDaemonCoreDutyCycle)
        ],
     "NEGOTIATOR": [
       (10, getRecentDaemonCoreDutyCycle),
       (30, lastNegotiationCycleTime0),
       # ... etc
     ],
    }
}

Reply:
Where the second elements are functions to call (probably passed the ClassAd of the job and returning a float between 0 and 1, inclusive).